pkg/apis/tensorflow/v1alpha1/defaults.go

```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/tensorflow/k8s/pkg/util"
+	"k8s.io/api/core/v1"
+)
+
+// SetDefaults sets any unspecified values to defaults
+func SetDefaults_TfJobSpec(c *TfJobSpec) error {
+	if c.TfImage == "" {
+		c.TfImage = DefaultTFImage
+	}
+
+	// Check that each replica has a TensorFlow container.
+	for _, r := range c.ReplicaSpecs {
+		if r.Template == nil && r.TfReplicaType != PS {
+			return fmt.Errorf("ReplicaType: %v, Replica is missing Template; %v", r.TfReplicaType, util.Pformat(r))
+		}
+
+		if r.TfPort == nil {
+			r.TfPort = proto.Int32(TfPort)
+		}
+
+		if string(r.TfReplicaType) == "" {
+			r.TfReplicaType = MASTER
```

Comment:
IMHO, should this be required? and `MASTER` is only one in distributed TF training.

Reply:
+1

Reply:
@wackxu this should fix